### PR TITLE
Mark incomplete periods

### PR DESCRIPTION
**Pull request by otfrom**
And experiment with adding transformation report to an existing excel with charts and pivot tables.

### CODE REVIEW

File: src/witan/cic/episodes/transformation_report.clj [hunk 1]
```diff
@@ -13,52 +13,90 @@
...
-                                 (::wcie/file-name new) (::wcie/sheet-name new) (::wcie/row-index new)
-                                 "Original" (str (::wce/report-date previous)) (str (::wce/ceased previous)) (::wce/placement previous) (::wce/legal-status previous) (::wce/care-status previous)]))
+  (let [template (xl/load-workbook "template-transformation-report.xlsx")
+        sheet (xl/select-sheet "Transformation Report" template)]
+    (xl/add-rows! sheet
+                  (x/into []
+                          format-rows-xf
+                          episodes))
+    (xl/save-workbook! out-file template))
+  #_(xl/save-workbook!
```
Comment (seb231) [inline src/witan/cic/episodes/transformation_report.clj:77]: ?

File: src/witan/cic/episodes/transformation_report.clj [hunk 2]
```diff
@@ -13,52 +13,90 @@
...
+               (str (::wce/report-date new)) (str (::wce/ceased new)) (::wce/placement new) (::wce/legal-status new) (::wce/care-status new)
+               (::wcie/file-name new) (::wcie/sheet-name new) (::wcie/row-index new)
+               "Original" (str (::wce/report-date previous)) (str (::wce/ceased previous)) (::wce/placement previous) (::wce/legal-status previous) (::wce/care-status previous)]))
+
+      (::wce/replacement edit)
+      (let [replacement (::wce/replacement edit)]
+        (into fields [(user-friendly-command-string (::wce/command edit)) (::wce/reason edit)
+                      (::wce/report-year replacement)
+                      (str (::wce/report-date replacement)) (str (::wce/ceased replacement)) (::wce/placement replacement) (::wce/legal-status replacement) (::wce/care-status replacement)
+                      (::wcie/file-name replacement) (::wcie/sheet-name replacement) (::wcie/row-index replacement)]))
```
Comment (seb231) [inline src/witan/cic/episodes/transformation_report.clj:58]: this formatting I find very confusing

Comment (seb231) [reply src/witan/cic/episodes/transformation_report.clj:58]: Is it just sorting and organising?

**Review by seb231 — APPROVED** [review]
Some quite complex fns strung together, but if you're happy I am

File: src/witan/cic/episodes/transformation_report.clj (hunk 1 above)
Comment (otfrom) [reply src/witan/cic/episodes/transformation_report.clj:77]: that is the old save workbook function and the new one. I'm not sure which I want to go with atm. I'd like to be able to template the workbook, but I'm not able to make that work atm.

File: src/witan/cic/episodes/transformation_report.clj (hunk 2 above)
Comment (otfrom) [reply src/witan/cic/episodes/transformation_report.clj:58]: I'll come back to this again and see if I can make it prettier later.

**Pull request merged**